packages/machine/test/integration/test-io-provider.ts

```diff
@@ -34,70 +31,30 @@ export class TestIOProvider {
       serializedMessage
     ) as cf.legacy.node.ClientActionMessage;
 
-    let done = false;
-    const executedListeners = [] as number[];
-    let count = 0;
-
-    // Invoke all listeners waiting for a response to resolve their promise
-    this.listeners.forEach(listener => {
-      if (
-        listener.appId === message.appId ||
-        (!listener.appId && listener.multisig === message.multisigAddress)
-      ) {
-        listener.method(message);
-        done = true;
-        executedListeners.push(count);
-        count += 1;
-      }
-    });
-
-    // Now remove all listeners we just invoked
-    executedListeners.forEach(index => this.listeners.splice(index, 1));
-
-    // initiate ack side if needed
-    if (message.seq === 1) {
+    if (this.waitForIoContinuation !== undefined) {
+      const k = this.waitForIoContinuation;
```

Comment:
Can we give this variable a better name?

Reply:
The name is explained in the PR. When not undefined this variable is the continuation (https://en.m.wikipedia.org/wiki/Continuation) of the call to waitForIo

Reply:
Right, but it's probably better if it's explained here as the code is the source of the truth.

Cross-referencing with the PR that merged the code adds unrealistic overhead to reach the explanation.

Reply:
added a comment